app/carbonapi/api.go

```diff
@@ -245,42 +247,42 @@ func registerPromehteusMetrics(logger *zap.Logger) {
 	}()
 }
 
-func loadBlockRuleHeaderConfig(ticker *time.Ticker, logger *zap.Logger) {
+func loadBlockRuleHeaderConfig(ticker *time.Ticker, logger *zap.Logger, app *App) {
 	var ruleConfig RuleConfig
 	for range ticker.C {
-		fileData, err := loadBlockRuleConfig()
+		fileData, err := loadBlockRuleConfig(app)
 
 		if err == nil {
 			err = yaml.Unmarshal(fileData, &ruleConfig)
 			if err != nil {
 				logger.Error("couldn't unmarshal block rule file data")
 			} else {
-				config.blockHeaderRules = ruleConfig
+				app.blockHeaderRules = ruleConfig
 			}
 		} else {
-			config.blockHeaderRules = RuleConfig{}
+			app.blockHeaderRules = RuleConfig{}
 		}
 	}
 }
 
-func loadBlockRuleConfig() ([]byte, error) {
+func loadBlockRuleConfig(app *App) ([]byte, error) {
```

Comment:
like elsewhere, I think makes sense to pass it only `app.config.BlockHeaderFile` here, to keep it clear that this function is fairly self-contained